### PR TITLE
chore: update CODEOWNERS with more specific permissions

### PR DESCRIPTION
**Pull request by Racer159**
This updates the `CODEOWNERS` file to protect some files but also open up more of the repository to others.

### CODE REVIEW

File: CODEOWNERS
Comment (mjnagel) [inline CODEOWNERS:1]: Most of these groups are missing or don't have write access - should we rectify that before merging?

Comment (mjnagel) [reply CODEOWNERS:1]: Might also be good to add the leapfrog team here? Not sure if uds-engine and/or uds-marketplace would be good as well?

Comment (Racer159) [reply CODEOWNERS:1]: Yep will def do that - want to get agreement these are the right (write :sweat_smile:) groups first - good callout (forgot them)

Comment (Racer159) [reply CODEOWNERS]: done

**Review by brandtkeller — APPROVED** [review]
Looks good pending repo setting modification for CODEOWNER permissions for file being valid.

**Review by corang — APPROVED** [review]

File: CODEOWNERS
Comment (mjnagel) [inline CODEOWNERS:4]: Probably makes sense to put `@defenseunicorns/leapfrogai` here too right?

**Review by mjnagel — APPROVED** [review]

File: CODEOWNERS
Comment (Racer159) [reply CODEOWNERS:4]: I left it to the teams / people I know have regularly fought with renovate - we could add them but wanted to limit slightly for now.

**Pull request merged**
